tdigest.go

```diff
@@ -76,7 +73,15 @@ func (t *TDigest) AddCentroidList(c CentroidList) {
 }
 
 // AddCentroid adds a single centroid.
+// Weights which are not a number or are <= 0 are ignored, as a NaN means.
 func (t *TDigest) AddCentroid(c Centroid) {
+	if math.IsNaN(c.Mean) {
+		return
+	}
+	if c.Weight <= 0 || math.IsNaN(c.Weight) || math.IsInf(c.Weight, 1) {
```

Comment:
Should these be combined into the same `if ... { return } ` block?

Reply:
Sure, I kept them apart because one is checking `Mean`, the other is checking `Weight`, and the line was getting long.